File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/vertx/repository/AllowedResultsRepository.java
package ru.tinkoff.kora.database.common.annotation.processor.vertx.repository;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import ru.tinkoff.kora.common.Mapping;
import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.common.annotation.processor.entity.TestEntityRecord;
import ru.tinkoff.kora.database.common.annotation.processor.vertx.VertxEntity;
import ru.tinkoff.kora.database.vertx.VertxRepository;

import java.util.List;

@Repository
public interface AllowedResultsRepository extends VertxRepository {
    @Query("INSERT INTO test(test) VALUES ('test')")
    Mono<Void> returnVoid();

    @Query("SELECT test")
    Mono<TestEntityRecord> returnObject();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapper.class)
    Mono<TestEntityRecord> returnObjectWithRowMapper();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapperNonFinal.class)
    Mono<TestEntityRecord> returnObjectWithRowMapperNonFinal();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapper.class)
    Mono<List<TestEntityRecord>> returnListWithRowMapper();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapperNonFinal.class)
    Mono<List<TestEntityRecord>> returnListWithResultSetMapper();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapper.class)
    Flux<TestEntityRecord> returnObjectFluxWithRowMapper();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapperNonFinal.class)
    Flux<TestEntityRecord> returnObjectFluxWithRowMapperNonFinal();


    @Query("INSERT INTO test(test) VALUES ('test')")
    void returnVoidBlocking();

    @Query("SELECT test")
    TestEntityRecord returnObjectBlocking();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapper.class)
    TestEntityRecord returnObjectWithRowMapperBlocking();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapperNonFinal.class)
    TestEntityRecord returnObjectWithRowMapperNonFinalBlocking();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapper.class)
    List<TestEntityRecord> returnListWithRowMapperBlocking();

    @Query("SELECT test")
    @Mapping(VertxEntity.TestEntityVertxRowMapperNonFinal.class)
    List<TestEntityRecord> returnListWithResultSetMapperBlocking();


}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/vertx/repository/AllowedResultsRepository.kt
package ru.tinkoff.kora.database.symbol.processor.vertx.repository

import ru.tinkoff.kora.common.Mapping
import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.symbol.processor.entity.TestEntity
import ru.tinkoff.kora.database.symbol.processor.vertx.TestEntityVertxRowMapper
import ru.tinkoff.kora.database.symbol.processor.vertx.TestEntityVertxRowMapperNonFinal
import ru.tinkoff.kora.database.symbol.processor.vertx.TestEntityVertxRowMapperNonNull
import ru.tinkoff.kora.database.vertx.VertxRepository

@Repository
interface AllowedResultsRepository : VertxRepository {
    @Query("INSERT INTO test(test) VALUES ('test')")
    fun returnVoid()

    @Query("SELECT test")
    fun returnPrimitive(): Int

    @Query("SELECT test")
    fun returnObject(): Int?

    @Query("SELECT test")
    fun returnNullableObject(): Int?

    @Query("SELECT test")
    @Mapping(TestEntityVertxRowMapper::class)
    fun returnObjectWithRowMapper(): TestEntity?

    @Query("SELECT test")
    @Mapping(TestEntityVertxRowMapperNonFinal::class)
    fun returnObjectWithRowMapperNonFinal(): TestEntity?

    @Query("SELECT test")
    @Mapping(TestEntityVertxRowMapper::class)
    fun returnOptionalWithRowMapper(): TestEntity?

    @Query("SELECT test")
    @Mapping(TestEntityVertxRowMapperNonNull::class)
    fun returnListWithRowMapper(): List<TestEntity>
}
